internal/codespaces/api/api.go

```diff
@@ -477,6 +478,73 @@ func (a *API) GetCodespacesMachines(ctx context.Context, repoID int, branch, loc
 	return response.Machines, nil
 }
 
+// TODO - should this be/is this in a more general API?
+func (a *API) GetCodespaceRepoSuggestions(ctx context.Context, partialSearch string) ([]string, error) {
+	reqURL := fmt.Sprintf("%s/search/repositories", a.githubAPI)
+	req, err := http.NewRequest(http.MethodGet, reqURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	parts := strings.Split(partialSearch, "/")
+	partsLen := len(parts)
+	if partsLen > 2 {
+		return nil, errors.New("repository name cannot have multiple slashes")
```

Comment:
Let's use `SplitN(str, "/", 2)` and thus not raise any error if there is more than one `/` character. We should refrain from perform repo name validation in the client in general. We just have to perform a different lookup based on whether there is a full owner component of the name or not

Reply:
In that case, the new strings.Cut/LastCut functions are your friend.